networkx/algorithms/tests/test_lowest_common_ancestors.py

```diff
@@ -51,38 +51,38 @@ def assert_has_same_pairs(d1, d2):
         for (a, b) in ((min(pair), max(pair)) for pair in chain(d1, d2)):
             assert get_pair(d1, a, b) == get_pair(d2, a, b)
 
-    def test_tree_all_pairs_lowest_common_ancestor1(self):
+    def test_specifying_root_optional_in_lca(self):
```

Comment:
I have started work on the changes and a quick question:
Since the test method is actually testing that adding a starting root produces the same result as not adding a root, won't this be more descriptive?
```suggestion
    def test_tree_all_pairs_lca_root_is_optional(self):
```

Reply:
I'd probably call it `lca_default_root`, but honestly it's quite subjective!

Remember - the goal is that the test names are informative so that when they fail there is some indication of what the test pertains to. A useful exercise to evaluate this is to run `pytest` with the verbose option, e.g. `pytest -v networkx/algorithms/tests/test_lowest_common_ancestors.py`